### DISCUSSION TITLE
Welcome to sile Discussions!

### DISCUSSION THREAD

**Discussion by alerque** [sile-typesetter/sile#1251] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by OlivierNicole** [comment]
Hello, I'm a computer science engineer with an interest in typography. It started when I used LaTeX for the first time to typeset a high school memoir, and never really stopped since.

I became interested in SILE because it takes as input a sane description language and can be programmed in a sane language — even though I like statically typed languages better, I can see how the run-time nature of everything in Lua helps to flexibly modify part of the typesetter during typesetting itself.

During the various lockdowns I worked on math typesetting in SILE because that's what I can most relate to, and I also hope that it might expand the interest in SILE to new people and maybe eventually draw more contributions to all aspects of SILE.

**Comment by Omikhleia** [comment]
Greetings, and thanks @alerque for the initiative opening these "Discussions" !

I am a French computer science engineer, born in the early 1970s, with a background in project management and software development (mostly in C++ and, these days, in JavaScript / TypeScript). I always enjoyed coding and hacking (in the noble sense). In the 1980s, my father introduced me to LaTeX, and by the early 1990s I was somehow quite good at it (maybe not a real wizard, but I had some decent knowledge of TeX's internals etc.). But I forgot most of this stuff as time passed... and the TeX world (from a end-user viewpoint) sort of lost me on the way, with all these Dvipxxx variants, then the OmegaTeX, pdfTeX, XeTeX, LuaTeX and Co... Doh! My head hurts.

I have an interest in typography and typesetting (I did typeset a few books of essays with Open/LibreOffice...) and typesetting systems of the kind of SILE: Beyond TeX/LaTeX, I tried Lout many years ago (Lol, [this one](https://en.wikipedia.org/wiki/Lout_(software)) is nearly forgotten, I guess? Weird beast...), and a bit of XSLT + XSL-FO and FOP (Erm... Oh my! Let's not comment here...) - So when I saw some announcement about SILE a few years ago, I thought I should give it a try... But learning Lua wasn't in my plans, so I postponed... Well, I started playing Minetest at some point (_smile_), and the scripting there is in Lua, so here I am eventually...

Oh, and I am (or used to be) a huge Tolkien fan, and also a casual writer (as a private hobby)... Just mentioning these things, because most of my current "tests" with SILE use stuff I have in a form or another, so you'll necessarily notice that _lineage_, if I am to provide examples in issues etc. :smiley_cat: 

As of my involvement in this community - Frankly, I dunno what to say, I just want to try typesetting cool stuff (that is, to my aesthetic taste) with less pain than with the aforementioned tools... And SILE has plenty of appealing qualities. So maybe I am just here to experiment around, and bother you all with patches, suggestions or questions, hehe.

**Reply by Omikhleia** [reply]
About 9 months after my initial dabbling into SILE, I have surely bothered this project with weird experiments, questions, suggestions and even casual complaints... Eh! It seems to be the good moment for an update :)

The [following book](https://sites.google.com/site/dragonbrumeux/contes-et-l%C3%A9gendes-dalmaq), published via Lulu.com POD services, is a 362-page novel in French, in the fiction/fantasy genre (in a broad sense).

The above page is in French too, obviously, but you may scroll down to the bottom and see a few pictures.

The interior content is entirely typeset with SILE (that is, all excepted the cover material).

While this is copyrighted material, of sorts, the decision was taken to share the whole SILE sources of the book in my `omikhleia-sile-packages` repository (examples folder), with the idea to illustrate how I used SILE (0.12.5) to edit the material and make that book a reality, relying on lots of features provided by the standard version of the software, and also (almost) all the packages I created on that journey -- So interested SILE friends could possibly have a look at the complete sources, would they want to see how the typesetting needs were addressed (sometimes in neat ways, but sometimes too -This editor counts on your indulgence- with dubious last-minute "hacks")

Disclaimers:
- Reproduction and/or distribution of the PDF or any derivative work in any form (including modified versions or printed copies) is NOT allowed without proper authorization. (One may obviously build a PDF from the sources, as long as it is kept as a private copy.)
- More importantly, this is not an advertisement not an incentive for anyone to buy the book. On the contrary, I will even suggest you should not do such a thing, unless you are really sure this is a kind of things you would enjoy reading...

The idea is that whenever someone asks if any actual books were typeset with SILE, you can now add another to the count ! :book:

**Reply by OlivierNicole** [reply]
That is extremely impressive :) and looks neat! I can only admire your determination—for the typesetting of course, but also for the book contents which looks quite rich!

**Reply by OlivierNicole** [reply]
I unfortunately haven't found it in me to keep hacking in SILE, partly because I lack of a typesetting project such as yours. But, maybe, one day…

**Reply by alerque** [reply]
This looks like a fantastic project—well done. However many hacks went into at the end of the day the output looks great and the effort has surely helped drive SILE forward so others won't need so many hacks. I with my French would let me read it, but I think I'll try to get my hands of a copy anyway just to have on the shelf to show off!

Also if you want to submit a guest blog post to the SILE website repo with a couple paragraphs about the project and the experience I'd be happy to have something other than boring release announcements to publish and spice things up over there ;-)

**Reply by simoncozens** [reply]
I also bought a copy of the book (just because I do like to have copies of anything typeset in SILE). It is quite amazing! As well as the wonderful world-building, I keep leafing through thinking "How did he do that?" :-)

**Reply by ctrlcctrlv** [reply]
I too wish I spoke French well enough to read it. Should have spent more time learning useful languages and not Esperanto :hand_over_mouth:

**Reply by Omikhleia** [reply]
And nearly one year after I started dabbling into SILE, where am I now?
- I am glad to tell that all of my own "main" classes and packages have been upgraded from SILE 0.12 to SILE 0.14 and now provided as installable via [luarocks](https://luarocks.org/m/sile) -- Call me an early adopter of the new "3rd party" package manager that was initiated in 0.13, but it rocks, so to speak, and it means anyone can give them a try more easily - It's pretty cool, with **9** "modules" being available.
- I still have 3 experimental packages and 3 specialized components needing the same type of upgrade... These will have to wait (it took 2h to 4h for each single file in the above-mentioned packages, over an intensive period of  3 weeks, so these - who need to be reconsidered anyway - are going to need more thorough evaluation).
- About the "main" packages, it goes beyond mere version upgrading and package bundling. For instance, in my previous book, I was never really happy with the page layout... So my book class now proposes a wide choice of them: the three 'french" layouts ("Canon des ateliers"), each with 3 additional common variants; two historical layouts (Vencentinus, Honnecourt); three "modern" variants of these; and - yes, why not, even Jan Tschichold's golden canon! So that's a choice amongst **18** proposed layouts:
   ![image](https://user-images.githubusercontent.com/18075640/187799820-25344c51-cd1c-44b8-9510-f381679aee95.png)

Where I am heading now...
- None of these page layouts properly defines where page headers and folios should go. Bringhurst, you won't see me often say this, is not very helpful here... So I am reading other obscure sources and a comparison of books, to check how this could also be proposed with a decent and sound variety of choices... The path to success is uncertain here!
- None of these "usual" book page layouts works well with side nodes (= margin notes; but read further), their margins are too tight. Here again, back on reading and experimenting the maths... I do see some interesting ways to do it while respecting the "canons", so there's a good chance it will see the light.

And of course, related to the last point, I am considering sides notes - well, more precisely, "stacked" side notes, just because they have my visual preference... Quite easy, in theory, but all things take time...

And it is _possible_ that my next book typesetting will require a square page size... An occasion to further the study of page layouts, if time permits it^^

**Reply by Omikhleia** [reply]
Greetings, fellow SILE users.
I am real glad to announce the second book I composed with SILE, a small [collection of essays on Tolkien's Middle-earth](https://sites.google.com/site/dragonbrumeux/on-some-stars-flowers-places-in-middle-earth).
This time, beside the regular print edition, the PDF is also freely available; and all sources (mostly in Djot, with but a bit of Markdown and SILE-Lua) are provided.

**Reply by OlivierNicole** [reply]
It looks very nice!

**Reply by Omikhleia** [reply]
Greetings friends,

Le Dragon de Brume did it again, with _On cartography, maps & locations in Middle-earth,_ another [collection of essays on Tolkien's Middle-earth](https://sites.google.com/site/dragonbrumeux/on-cartography-maps-locations-in-middle-earth) translated from French! As for the previous volume, besides the print-on-demand edition, the PDF and all source files are also freely available under appropriate licenses.

I'm very happy with this new book -- it's my third "real book" composed with SILE, and most of all the second one using my "resilient" framework -- with sources mostly in Djot, with still a bit a SILE-Lua here and there.

My only (small) regret is that I was too short in time to update my whole setup, so the book was generated with the latest resilient modules (updated for that purpose) _but however_ still with a custom build of SILE 0.14 and a few hacks here and there...
There's a proverb in French "Jamais deux sans trois" (which might perhaps translate as "Good things come in threes" ?) -- So I do hope there will be another Tolkien-related volume published by Le Dragon de Brume, and that time, surely, it will use an updated setup... Apart from that, things went smoothly -- "resilient" is getting mature and SILE itself has far less issues than when I started using it seriously, back in Fall 2021.

**Reply by Omikhleia** [reply]
I had to move on quicker than expected for various reasons... So Le Dragon de Brume just published another book again, available online: [_A bibliography of Tolkien studies in French & English_](https://sites.google.com/site/dragonbrumeux/bibliography) -- based on the in-development modules for extended bibliographies with CSL support (#2082), still unfinished at this date and thus used here with a few "quick'n dirty" hacks... Yet, I'm very happy with the "initial" result, and -- at least in my opinion -- it shows the path forward, so the next edition may eventually rely on something more robust, typographically nicer, and having passed some kind of "test of fire".
As usual, SILE (even using some still in-development experimental features here) did great! Oh, and I finally used SILE 0.15, so my next steps would now be to finish all those things (that is, the bibliography support for SILE, and also officially migrating all my 3rd-party packages to 0.15; and then add a few extra things to my "resilient" framework...).
As professor Tolkien said, "the road goes even on" -- that is, there's always a new adventure awaiting...

**Reply by Omikhleia** [reply]
Greetings.
I could say a lot about my current projects, book updates, new planned books, potential contributions, etc.
But these, though soon-coming for some of them, will have to wait, at least a day.
It's close but not yet midnight here in my area. So...
Today, to the day, is the anniversary of my very first issue opened on the SILE repository on Aug. 2021, 29; and when I met you all here.
So I just want to thank the people from these early days, and onward to the future: @simoncozens for having imagined SILE, as a dream to pursue; @alerque for maintaining it, with all his always astonishing skills in many varied areas; @ctrlcctrlv whom I miss here (and who also does great fonts, go check them); @OlivierNicole who made SILE's math possible; and many other people on the journey along. Thanks.

Did I ever explained how I found SILE, actually? Maybe not, or forgive me otherwise...
I had these short fantasy stories I wanted to assemble in a book, related to the fictional "land of Silé", imagined many years ago. Having lost some of my old material due to a past theft, I google'd for lost stories (which I might had given to arcane, and now defunct, web sites), just in case I could recover some texts I'd even forgotten, using many keywords. Including _Silé_, the land where resides the city of Almaq. Guess what? I didn't know there was a city by that name in Turkey... Er... No, wrong story :)
Back on SILE, so...
Unknowingly I got something else than expected, something more: the typesetting solution that would make that dreamed book possible. That day my eyes shined. I've tried a lot of typesetting solutions, and will try more, certainly.
But the light in my eyes on that day will be hard to match.
Call that a sign in the skies, or whatever else you prefer.
So... Thanks!

**Reply by Omikhleia** [reply]
The new edition of [_A bibliography of Tolkien studies in French & English_](https://sites.google.com/site/dragonbrumeux/bibliography) (Summer 2025 edition) is live -- not only the PDF booklet is composed with SILE as before, but the online web edition was also generated using SILE... Well, sort of, i.e. using SILE as a "Lua-on-steroids" toolkit.
